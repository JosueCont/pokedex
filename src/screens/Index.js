import React, {useEffect, useContext, useState} from 'react';
import {View, Text, StyleSheet, TextInput, FlatList} from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';
import ListPokemon from '../components/ListPokemon';
import {Context as PokemonContext} from '../context/pokemonsApi';


const Index = ({navigation}) => {
  const useData = () => {
    return !!state.data.results && state.data.results.filter(pokemon=>{
      return  pokemon.name.toLowerCase().includes(searchPokemon.toLowerCase())
     })     
  }
  useEffect(() => {
    fecthData();
  }, []);
  const {state, fecthData} = useContext(PokemonContext);
  const [searchPokemon, setSearchPokemon] = useState('');
  console.log('losDAtos', state.data.results);
  console.log('buscar',searchPokemon);
  return (
    <View style={styles.screen}>
      <View style={styles.position}>
        <TextInput
          placeholder="Buscar pokemon"
          style={styles.input}
          value={searchPokemon}
          onChangeText={e => setSearchPokemon(e)}
        />
        <Icon name="search" size={20} color="#000" style={styles.icon} />
      </View>
       <FlatList
          contentContainerStyle={styles.container}
          numColumns={2}
          data={useData()}
          renderItem={({item,key}) => (
            <ListPokemon
              key={key}
              item={item}
              navigator={() => navigation.navigate('Detalles', {item})}
            />
          )}
          keyExtractor={item => item.name}
        />
     
    </View>
  );
};

const styles = StyleSheet.create({
  screen: {
    backgroundColor: 'white',
    flex: 1,
    paddingTop: 10,
  },
  position: {
    position: 'relative',
  },
  input: {
    width: 360,
    height: 40,
    alignSelf: 'center',
    borderWidth: 1,
    borderColor: 'gray',
    paddingHorizontal: 25,
    borderRadius: 10,
    marginBottom: 20,
  },
  icon: {
    position: 'absolute',
    top: 10,
    left: 30,
  },
  container: {
    flexDirection: 'column',    
    alignItems: 'center',
    marginHorizontal: 15,
  },
});

export default Index;
