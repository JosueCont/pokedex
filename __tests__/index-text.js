/**
 * @format
 */

import 'react-native';
import renderer from 'react-test-renderer';
//import React,{useState} from 'react';
import {TouchableOpacity} from 'react-native';
import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import {shallow, mount} from 'enzyme';
import Index from '../src/screens/Index';
Enzyme.configure({ adapter: new Adapter() });

// Note: test renderer must be required after react-native.
describe('index screen', () =>{
  it('renders correctly', () => {
    jest.mock('react-native-vector-icons/FontAwesome', () => 'Icon')
    const snap= renderer.create(<Index  />).toJSON();
    expect(snap).toMatchSnapshot();
  });
  test('Press the button to each pokemon',()=>{
    const onPressMock = jest.fn();
    const wrapper= shallow(<TouchableOpacity onPress={ onPressMock} />);
    wrapper.find('TouchableOpacity').simulate('Press')
    expect(onPressMock).toHaveBeenCalled();
    expect(onPressMock.mock.calls.length).toEqual(1);  
  });
})

