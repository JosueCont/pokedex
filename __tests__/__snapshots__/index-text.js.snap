// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index screen renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "paddingTop": 10,
    }
  }
>
  <View
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      placeholder="Buscar pokemon"
      rejectResponderTermination={true}
      style={
        Object {
          "alignSelf": "center",
          "borderColor": "gray",
          "borderRadius": 10,
          "borderWidth": 1,
          "height": 40,
          "marginBottom": 20,
          "paddingHorizontal": 25,
          "width": 360,
        }
      }
      underlineColorAndroid="transparent"
    />
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 20,
          },
          Object {
            "left": 30,
            "position": "absolute",
            "top": 10,
          },
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      ï€‚
    </Text>
  </View>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "flexWrap": "wrap",
        "marginHorizontal": 15,
      }
    }
    data={
      Array [
        Object {
          "name": "pikachu",
          "number": "1",
        },
        Object {
          "name": "Squirtle",
          "number": "2",
        },
        Object {
          "name": "charmander",
          "number": "3",
        },
        Object {
          "name": "Bolvasur",
          "number": "4",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "borderRadius": 8,
              "borderWidth": 1,
              "height": 200,
              "marginBottom": 10,
              "marginRight": 10,
              "opacity": 1,
              "paddingLeft": 10,
              "paddingTop": 10,
              "width": 180,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "green",
                "fontSize": 15,
                "marginBottom": 10,
              }
            }
          >
            #
            1
          </Text>
          <Image
            alt="Imagen pokemon"
            source={
              Object {
                "testUri": "../../../src/assets/logo192.png",
              }
            }
            style={
              Object {
                "alignSelf": "center",
                "height": 100,
                "width": 100,
              }
            }
          />
          <Text
            style={
              Object {
                "alignSelf": "center",
                "fontSize": 20,
                "marginTop": 10,
              }
            }
          >
            pikachu
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "borderRadius": 8,
              "borderWidth": 1,
              "height": 200,
              "marginBottom": 10,
              "marginRight": 10,
              "opacity": 1,
              "paddingLeft": 10,
              "paddingTop": 10,
              "width": 180,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "green",
                "fontSize": 15,
                "marginBottom": 10,
              }
            }
          >
            #
            2
          </Text>
          <Image
            alt="Imagen pokemon"
            source={
              Object {
                "testUri": "../../../src/assets/logo192.png",
              }
            }
            style={
              Object {
                "alignSelf": "center",
                "height": 100,
                "width": 100,
              }
            }
          />
          <Text
            style={
              Object {
                "alignSelf": "center",
                "fontSize": 20,
                "marginTop": 10,
              }
            }
          >
            Squirtle
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "borderRadius": 8,
              "borderWidth": 1,
              "height": 200,
              "marginBottom": 10,
              "marginRight": 10,
              "opacity": 1,
              "paddingLeft": 10,
              "paddingTop": 10,
              "width": 180,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "green",
                "fontSize": 15,
                "marginBottom": 10,
              }
            }
          >
            #
            3
          </Text>
          <Image
            alt="Imagen pokemon"
            source={
              Object {
                "testUri": "../../../src/assets/logo192.png",
              }
            }
            style={
              Object {
                "alignSelf": "center",
                "height": 100,
                "width": 100,
              }
            }
          />
          <Text
            style={
              Object {
                "alignSelf": "center",
                "fontSize": 20,
                "marginTop": 10,
              }
            }
          >
            charmander
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "borderRadius": 8,
              "borderWidth": 1,
              "height": 200,
              "marginBottom": 10,
              "marginRight": 10,
              "opacity": 1,
              "paddingLeft": 10,
              "paddingTop": 10,
              "width": 180,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "green",
                "fontSize": 15,
                "marginBottom": 10,
              }
            }
          >
            #
            4
          </Text>
          <Image
            alt="Imagen pokemon"
            source={
              Object {
                "testUri": "../../../src/assets/logo192.png",
              }
            }
            style={
              Object {
                "alignSelf": "center",
                "height": 100,
                "width": 100,
              }
            }
          />
          <Text
            style={
              Object {
                "alignSelf": "center",
                "fontSize": 20,
                "marginTop": 10,
              }
            }
          >
            Bolvasur
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
